providers/implementations/rands/seeding/rand_unix.c

```diff
@@ -695,7 +715,7 @@ size_t prov_pool_acquire_entropy(RAND_POOL *pool)
         return entropy_available;
 #   endif
 
-#   if defined(OPENSSL_RAND_SEED_EGD)
+#   if defined(OPENSSL_RAND_SEED_EGD) && !defined(FIPS_MODULE)
```

Comment:
Why the extra check after explicitly undef‘ing EGD?

(same question for the other defines)

Reply:
There is no need for the extra check, I've removed it.